src/qibolab/platforms/abstract.py

```diff
@@ -16,6 +17,85 @@
 )
 
 
+class Qubit:
+    """Data structure that holds information about all instruments controlling a qubit.
+
+    Args:
+        index (int): Qubit index.
+        settings (dict): Settings dictionary as read from the runcard yaml.
+
+    Attributes:
+        instruments (list): List of the three instrument names (str) controlling this qubit.
+        channels (list): List of the three channels (int) associated to the qubit.
+        ro_channel, qd_channel, qf_channel (int): Explicit references to the channels.
+        ro_port, qd_port, qf_port: Explicit references to the ports.
+    """
+
+    def __init__(self, index, settings):
+        self.index = index
+        self.channels = settings["qubit_channel_map"][index]
+        self.ro_channel, self.qd_channel, self.qf_channel = self.channels
+
+        # Generate qubit_instrument_map from qubit_channel_map and the instruments' channel_port_maps
+        self.instruments = [None, None, None]
+        for name, value in settings.get("instruments", {}).items():
+            instrument_settings = value.get("settings")
+            channel_port_map = instrument_settings.get("channel_port_map", [])
+            s4g_modules = instrument_settings.get("s4g_modules", [])
+            for channel in itertools.chain(channel_port_map, s4g_modules):
+                if channel in self.channels:
+                    self.instruments[self.channels.index(channel)] = name
+
+        self.ro_port, self.qd_port, self.qf_port = None, None, None
+
+        self.native_one_qubit = settings["native_gates"]["single_qubit"][index]
+        # TODO: Think how to implement two qubit gates
+
+    def set_ports(self, instruments):
```

Comment:
I would have the functionality of this function within `__init__()` so that it is not necessary to call it.